kpTranslator/database/models.py

```diff
@@ -2,6 +2,22 @@
 from datetime import datetime
 from kpTranslator.config import db, ma
 
+
 class TestModel(db.Model):
     id = db.Column(db.Integer, primary_key=True)
     name = db.Column(db.String(80))
+
+
+class TranslatorKnowledgeGraphNode(db.Model):
```

Comment:
Is there a reason you're not storing all of the information in the file?

Reply:
For now, I'm only pulling columns from the TSV files that are considered essential for a Translator knowledge graph. I need to figure out how to define model classes that are partially specified at data load time. The idea is that nodes/edges for a knowledge graph will have a set of required properties (the ones I'm loading right now) and a set of additional properties, which could be just about anything, in theory. I'll be interested in your recommendations for defining some model/schema properties on-the-fly based on the column headers in the nodes and edges files. Do you know how I would do this?

Reply:
ModelSchemas are generally not very flexible to change.  You'd probably either
1) make a model that contains all the values and allow some to be null if they're optional
2) make a field that contains the 'rest of' the data and put all of the optional fields in there.  This could be something like a jsonb field in PostgreSQL - https://www.postgresql.org/docs/current/functions-json.html which could be a JSON1 extension in sqlite - https://www.sqlite.org/json1.html. If you go this path I'd recommend setting up a postgres container for local development rather than using JSON1 though.